src/prompt-input/styles.scss

```diff
@@ -95,14 +131,71 @@ $send-icon-right-spacing: awsui.$space-static-xxs;
     }
   }
 
-  &.textarea-invalid {
-    @include styles.form-invalid-control();
+  &-wrapper {
+    display: flex;
   }
+}
 
-  &.textarea-warning {
-    @include styles.form-warning-control();
+.button {
+  align-self: end;
+  padding-inline-end: calc(styles.$control-padding-horizontal / 2);
+  padding-block-end: awsui.$space-scaled-xxxs;
+
+  > .action-button {
+    padding: 0;
+
+    // offset the focus ring by 1px per side so it doesn't blend into the textarea border
+    @include focus-visible.when-visible {
+      @include styles.focus-highlight(
+        (
+          'vertical': calc((-1 * #{awsui.$space-xxxs}) - 1px),
+          'horizontal': calc((#{awsui.$space-xxxs}) - 1px),
+        )
+      );
+    }
   }
-  &.textarea-with-button {
-    padding-inline-end: calc(styles.$control-padding-horizontal + $send-icon-right-spacing + awsui.$size-icon-normal);
+}
+
+.secondary-content {
+  @include styles.styles-reset;
+  @include styles.control-border-radius-full();
+  padding-block-start: styles.$control-padding-vertical;
+  padding-block-end: awsui.$space-scaled-s;
```

Comment:
Shouldn't this padding also be conditional, i.e. inside the `.with-paddings` block?